test/alternator/test_encoding.py

```
@@ -309,3 +314,61 @@ def test_index_key_not_schema_column(dynamodb, cql, table_with_indexes):
+# A user may pre-create the keyspace "alternator_{name}" with CQL to override
+# the properties which Alternator would have picked for it. Check that a table
+# created afterwards indeed lands in that keyspace: the keyspace is created
+# with vnodes while the table's tag asks for tablets, so a table using vnodes
+# proves the pre-created keyspace was used.
+def test_precreated_keyspace(dynamodb, cql, this_dc):
+    name = unique_table_name()
+    with precreated_keyspace(cql, this_dc, name, tablets=False):
+        with new_test_table(dynamodb, name=name,
+                Tags=[{'Key': 'system:initial_tablets', 'Value': '0'}],
+                KeySchema=[{'AttributeName': 'p', 'KeyType': 'HASH'}],
+                AttributeDefinitions=[{'AttributeName': 'p', 'AttributeType': 'S'}]) as table:
```

Ruff (0.16.2)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)